service/worker/deployment/deployment_util.go

```diff
@@ -0,0 +1,226 @@
+		TaskQueue:                &taskqueuepb.TaskQueue{Name: primitives.PerNSWorkerTaskQueue},
+		Input:                    workflowInputPayloads,
+		WorkflowIdReusePolicy:    enumspb.WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE,
+		WorkflowIdConflictPolicy: enumspb.WORKFLOW_ID_CONFLICT_POLICY_USE_EXISTING,
+		SearchAttributes:         sa,
+	}
+
+	updateReq := &workflowservice.UpdateWorkflowExecutionRequest{
+		Namespace: d.namespaceEntry.Name().String(),
+		WorkflowExecution: &commonpb.WorkflowExecution{
+			WorkflowId: deploymentWorkflowID,
+		},
+		Request: &updatepb.Request{
+			Input: &updatepb.Input{Name: RegisterWorkerInDeployment, Args: updatePayload},
+		},
+		WaitPolicy: &updatepb.WaitPolicy{LifecycleStage: enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_COMPLETED},
+	}
+
+	// This is an atomic operation; if one operation fails, both will.
+	_, err = d.historyClient.ExecuteMultiOperation(ctx, &historyservice.ExecuteMultiOperationRequest{
+		NamespaceId: d.namespaceEntry.ID().String(),
+		WorkflowId:  deploymentWorkflowID,
+		Operations: []*historyservice.ExecuteMultiOperationRequest_Operation{
+			{
+				Operation: &historyservice.ExecuteMultiOperationRequest_Operation_StartWorkflow{
+					StartWorkflow: &historyservice.StartWorkflowExecutionRequest{
+						NamespaceId:  d.namespaceEntry.ID().String(),
+						StartRequest: startReq,
+					},
+				},
+			},
+			{
+				Operation: &historyservice.ExecuteMultiOperationRequest_Operation_UpdateWorkflow{
+					UpdateWorkflow: &historyservice.UpdateWorkflowExecutionRequest{
+						NamespaceId: d.namespaceEntry.ID().String(),
+						Request:     updateReq,
+					},
+				},
+			},
+		},
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GenerateDeploymentWorkflowID is a helper that generates a system accepted
+// workflowID which are used in our deployment workflows
+func (d *DeploymentWorkflowClient) GenerateDeploymentWorkflowID() string {
+	deploymentWorkflowID := DeploymentWorkflowIDPrefix + DeploymentWorkflowIDDelimeter + d.deployment.DeploymentName + DeploymentWorkflowIDDelimeter + d.deployment.BuildId
+	return d.EscapeChar(deploymentWorkflowID)
+}
+
+// GenerateStartWorkflowPayload generates start workflow execution payload
+func (d *DeploymentWorkflowClient) GenerateStartWorkflowPayload() (*commonpb.Payloads, error) {
+	workflowArgs := &deployspb.DeploymentWorkflowArgs{
+		NamespaceName:     d.namespaceEntry.Name().String(),
+		NamespaceId:       d.namespaceEntry.ID().String(),
+		DeploymentName:    d.deployment.DeploymentName,
+		BuildId:           d.deployment.BuildId,
+		TaskQueueFamilies: nil,
+	}
+	workflowInputPayloads, err := sdk.PreferProtoDataConverter.ToPayloads(workflowArgs)
+	if err != nil {
+		return nil, err
+	}
+	return workflowInputPayloads, nil
+}
+
+// GenerateUpdateDeploymentPayload generates update workflow payload
+func (d *DeploymentWorkflowClient) GenerateUpdateDeploymentPayload(taskQueueName string, taskQueueType enumspb.TaskQueueType,
```

Comment:
Meant to rename this too?

```suggestion
func (d *DeploymentWorkflowClient) generateRegisterWorkerInDeploymentArgs(taskQueueName string, taskQueueType enumspb.TaskQueueType,
```